src/clusterfuzz/_internal/system/archive.py

```
@@ -34,111 +36,316 @@
+    # If the output directory is a symlink, get its actual path since we will be
+    # doing directory traversal checks later when unpacking the archive.
+    output_directory = os.path.realpath(path)
+
+    # If the archive is not trusted, do file path checks to make
+    # sure this archive is safe and is not attempting to do path
+    # traversals.
+    if not trusted:
+      if (_is_attempting_path_traversal(self._zip_archive.filename,
+                                        output_directory, member)):
+        return None
+
     try:
-      with lzma.LZMAFile(archive_path) as lzma_file, \
-            tarfile.open(fileobj=lzma_file) as tar_file:
+      extracted_path = self._zip_archive.extract(member, output_directory)
+
+      # Preserve permissions for regular files. 640 is the default
+      # permission for extract. If we need execute permission, we need
+      # to chmod it explicitly. Also, get rid of suid bit for security
+      # reasons.
+      external_attr = self._zip_archive.getinfo(member).external_attr >> 16
+      if oct(external_attr).startswith(FILE_ATTRIBUTE):
+        old_mode = external_attr & 0o7777
+        new_mode = external_attr & 0o777
+        new_mode |= 0o440
+        needs_execute_permission = external_attr & 100
+
+        if new_mode != old_mode or needs_execute_permission:
+          # Default extract condition is 640 which is safe.
+          # |new_mode| might have read+write+execute bit for
+          # others, so remove those.
+          new_mode &= 0o770
+
+          os.chmod(extracted_path, new_mode)
+
+      # Create symlink if needed (only on unix platforms).
+      if (trusted and hasattr(os, 'symlink') and
+          oct(external_attr).startswith(SYMLINK_ATTRIBUTE)):
+        symlink_source = self._zip_archive.read(member)
+        if os.path.exists(extracted_path):
+          os.remove(extracted_path)
+        os.symlink(symlink_source, extracted_path)
+
+      return extracted_path
+    except:
```

Let's catch the exception and log something about it, instead of swallowing it silently.